driver/src/contracts/stablex_contract.rs

```diff
@@ -77,15 +79,15 @@ pub struct FilteredOrderPage {
 pub trait StableXContract {
     /// Retrieve the current batch ID that is accepting orders. Note that this
     /// is always `1` greater than the batch ID that is accepting solutions.
-    fn get_current_auction_index(&self) -> Result<u32>;
+    fn get_current_auction_index(&self) -> BoxFuture<'static, Result<u32>>;
```

Comment:
nit: Can this be '_? The reason I say this is that it makes the bounds less restrictive and is more resistant to changes in the implementation (which may end up borrowing self over the course of the `Future`.

Reply:
It can't be `'_` (doesn't compile in the trait) but it can be `fn foo<'a>(&'a self) -> BoxFuture<'a, Result<...>>` which makes the most sense to me.

Reply:
That is what I hoped the elided `'_` would mean, but I guess it doesn't work in this particular case.